installer/scripts/service_control

```diff
@@ -321,13 +324,27 @@ disable_omsagent_service()
     rm -f $CONF_DIR/.service_registered
 }
 
+check_ws_and_invoke()
+{
+    for ws_id in $NON_OMS_WS
+    do
+        ls -1 $ETC_DIR | grep -w ${ws_id} > /dev/null 2>&1
+        if [ $? -eq 0 ]; then
+            setup_variables ${ws_id}
+            $1
+        fi
+    done
+}
+
 start_all_omsagent()
 {
     for ws_id in `ls -1 $ETC_DIR | grep -E $WORKSPACE_REGEX`
     do
        setup_variables ${ws_id}
        start_omsagent
     done
+
+    check_ws_and_invoke start_omsagent
```

Comment:
Have you run a quick test with these cases just to make sure this works as intended:

1. only SCOM workspace
2. only OMS workspace
3. SCOM and OMS workspace

It looks good to me, but verification would be ideal.